Sming/SmingCore/Network/Http/HttpRequest.h

```diff
@@ -117,15 +193,38 @@ class HttpRequest
 		return responseStream;
 	}
 
-	HttpRequest* onHeadersComplete(RequestHeadersCompletedDelegate delegateFunction);
-	HttpRequest* onBody(RequestBodyDelegate delegateFunction);
-	HttpRequest* onRequestComplete(RequestCompletedDelegate delegateFunction);
+	HttpRequest* onHeadersComplete(RequestHeadersCompletedDelegate delegateFunction)
+	{
+		headersCompletedDelegate = delegateFunction;
+		return this;
+	}
+
+	HttpRequest* onBody(RequestBodyDelegate delegateFunction)
+	{
+		requestBodyDelegate = delegateFunction;
+		return this;
+	}
 
+	HttpRequest* onRequestComplete(RequestCompletedDelegate delegateFunction)
+	{
+		requestCompletedDelegate = delegateFunction;
+		return this;
+	}
+
+	/** @brief Clear buffers and reset to default state in preparation for another request */
 	void reset();
 
 #ifdef ENABLE_SSL
-	HttpRequest* setSslOptions(uint32_t sslOptions);
-	uint32_t getSslOptions();
+	HttpRequest* setSslOptions(uint32_t sslOptions)
+	{
+		sslOptions = sslOptions;
```

Comment:
Yep, that was the problem. Can you put a PR with the small fix? Sorry for not noticing this earlier. There should be a way to detect such problems automatically.

Reply:
I made the error moving the code into the header - yes, it would be useful to be able to track those sorts of changes in diff.

Reply:
@slaff Thank you!

Reply:
The issue is fixed with #1504. No action is needed from your side.

> yes, it would be useful to be able to track those sorts of changes in diff.

I guess my IDE, cpplint or codacy should be able to detect the issue. Will look tomorrow how to enable that in codacy by default.